applications/asset_tracker_v2/src/ext_sensors/ext_sensors_bsec.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include <sys/atomic.h>
+#include <spinlock.h>
+#include <settings/settings.h>
+#include <sys/byteorder.h>
+#include <drivers/i2c.h>
+
+#include "bsec_integration.h"
+#include "ext_sensors_bsec.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ext_sensors_bsec, CONFIG_EXTERNAL_SENSORS_LOG_LEVEL);
+
+BUILD_ASSERT(CONFIG_FP_HARDABI, "CONFIG_FP_HARDABI must be set when using the BME680 BSEC library");
+
+/* Sample rate for the BSEC library
+ *
+ * BSEC_SAMPLE_RATE_ULP = 0.0033333 Hz = 300 second interval
+ * BSEC_SAMPLE_RATE_LP = 0.33333 Hz = 3 second interval
+ * BSEC_SAMPLE_RATE_HP = 1 Hz = 1 second interval
+ */
+#if defined(CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_MODE_ULTRA_LOW_POWER)
+#define BSEC_SAMPLE_RATE BSEC_SAMPLE_RATE_ULP
+#elif defined(CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_MODE_LOW_POWER)
+#define BSEC_SAMPLE_RATE BSEC_SAMPLE_RATE_LP
+#elif defined(CONFIG_EXTERNAL_SENSORS_BSEC_SAMPLE_MODE_CONTINUOUS)
+#define BSEC_SAMPLE_RATE BSEC_SAMPLE_RATE_HP
+#endif
+
+/* Temperature offset due to external heat sources. */
+float temp_offset = (CONFIG_EXTERNAL_SENSORS_BSEC_TEMPERATURE_OFFSET / (float)100);
+
+/* @breif Interval for saving BSEC state to flash
+ *
+ * Interval = BSEC_SAVE_INTERVAL * 1/BSEC_SAMPLE_RATE
+ * Example:  600 * 1/0.33333 Hz = 1800 seconds = 0.5 hour
+ */
+#define BSEC_SAVE_INTERVAL 600
+
+/* Definitions used to store and retrieve BSEC state from the settings API */
+#define SETTINGS_NAME_BSEC "bsec"
+#define SETTINGS_KEY_STATE "state"
+#define SETTINGS_BSEC_STATE SETTINGS_NAME_BSEC "/" SETTINGS_KEY_STATE
+
+/* Stack size of internal BSEC thread. */
+#define BSEC_STACK_SIZE 4096
```

Comment:
This is such a large stack size that I think we can expose it as a Kconfig option to let users tune it if they want to